### PR TITLE
add products to cart and order

### PR DESCRIPTION
**Pull request by at-huydoan**
add product to cart.
update and remove cart item
make order

### CODE REVIEW

File: app/Http/Frontend/Controllers/CartController.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @param int $id id product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function buyProduct($id)
```
Comment (ghost) [inline app/Http/Frontend/Controllers/CartController.php:32]: function name `addProduct` is better than `buyProduct`

File: app/Http/Frontend/Controllers/CartController.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $products = Product::findOrFail($id);
```
Comment (ghost) [inline app/Http/Frontend/Controllers/CartController.php:34]: it's just one product, use single, not plural

File: app/Http/Frontend/Controllers/CartController.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        Cart::add(['id' => $id, 'name' => $products->name, 'qty' => 1, 'price' => $products->price, 'options' => ['image' => $products->image]]);
```
Comment (ghost) [inline app/Http/Frontend/Controllers/CartController.php:35]: this line is too long, it's not good, please separate to many lines

File: app/Http/Frontend/Controllers/CartController.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request for create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
```
Comment (ghost) [inline app/Http/Frontend/Controllers/CartController.php:46]: what is the purpose of this function ?

File: app/Http/Frontend/Controllers/CartController.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request for update
+     * @param int                      $id      id admin user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
```
Comment (ghost) [inline app/Http/Frontend/Controllers/CartController.php:63]: function name should be `updateQuantity`

**Comment by ghost** [comment]
@at-huyqd please separate to 2 or 3 pull requests, it's difficult to review. Thanks

**Comment by at-huydoan** [comment]
ok, i will separate them

File: app/Http/Frontend/Controllers/CartController.php (hunk 4 above)
Comment (at-huydoan) [inline app/Http/Frontend/Controllers/CartController.php:46]: add product to cart with quantity in detail product

**Pull request closed without merging**